firmware/arduino/chessboard/uart.ino

```diff
@@ -0,0 +1,62 @@
+char incomingByte;
+char  buffer[5];
+int byteNum = -1; //-1 indicates that the start code hasn't been received
+
+//Wait for input
+void serialEvent2()
+{
+    //Loop through all available bytes
+    while (Serial2.available())
+    {
+        //Get byte
+        incomingByte = (char) Serial2.read();
+
+        //Reset buffer position
+        if (incomingByte == '~')
+        {
+            byteNum = 0;
+        }
+        //Add byte to buffer
+        else if (byteNum != -1)
+        {
+            buffer[byteNum] = incomingByte;
+        }
+
+        //Check if the buffer is full, process input
+        if (byteNum == 4)
+        {
+            //Reset buffer position
+            byteNum = -1;
+
+            //Process input
```

Comment:
Per our other thread, here would be a good spot to call a helper function to parse the message. Something like `parseMessageFromPi(&buffer)` and maybe that can return a bool which you can use to tell whether the message was valid or not (maybe bytes got lost etc).

Reply:
I just did this, but I don't want to mark the conversation resolved until you've had a chance to look at it.

Reply:
Looks great